### DISCUSSION TITLE
No keyboard (or few) keyboard bindings working for SPICE session.

### DISCUSSION THREAD

**Discussion by nrvale0** [regolith-linux/regolith-desktop#1085] [Q&A]
When logging into a Debian 12 VM over SPICE I'm getting either no keyboard combos working or just Super+Enter for terminal working. Not even Super+R and Super+C. 

Interestingly, I created .desktop file and dropped in ~/.config/autostart which fires up xev in a terminal. That starts successfully and I can then verify that the keyboard combos are being received by the X11 sessions via the output of the xev window. This is with the i3/X11 session.

Anyone else?

Thoughts?

**Comment by nrvale0** [comment]
Interestingly, the same VM with i3 over SPICE works fine.

**Reply by nrvale0** [reply]
Ended up being a missed package install. Validating and documenting now. Current theory is that it was related to one of the x11 session packages...

**Reply by kgilmer** [reply]
Hmm, if you find the rootcause LMK if we're missing so dependency, thanks!

**Reply by nrvale0** [reply]
Well, I have bad news. Twice. I have what appears to be 2 red herrings...

1) I think that the missing packages were probably a results of my thrashing on the VM to try to get it working. When I create another VM and use the install instructions the full complement of packages are installed.

2) Between now and when I created this Discussion entry even =xev= has stopped working on these VM + SPICE sessions. It doesn't show any keyboard input. Nor mouse input (meaning entry/exit) either.

Behavior which remains consistent:

1) With default configs on various desktops + WMs, I am still able to do all of the usual keyboard input.

2) With the default regolith 3 X11 config I can still create terminals (Mod+Enter) and I can *sometimes* close them with Ctrl-d but that's it. No other keyboard input seems to be caught by the Regolith session.

I had a theory that perhaps something about the newest Proxmox version or about my usage of input-remapper on the local system was causing the issue but I have not found any evidence of that in testing.

I'll likely just fall back to other GUI shells for VM usage. If so, I'll close this out in a couple of days.

**Comment by mkettn** [comment]
Hi, I guess I have the same problem with debian 12 in a VM. Regolith doesn't work. It appears that theres a invisible window in front which sucks up all the keys. I can only see that the mouse sometimes changes its appearance.
however, only i3 works fine.
on LMDE at least the shortcut helper window will be shown. but here again the same problem.
I've never tried regolith outside of a VM. maybe I'll give it a shot on a real machine then.

**Comment by ristomatti** [comment]
I ahve the same exact experience on a Ubuntu 24.04 QEMU VM within Promox. Plain i3 desktop session works, but Regolith / X11 reacts pretty much only to mouse, meta+enter, meta+space and meta+shift+e to exit the session (and even that is really laggy).

**Reply by kgilmer** [reply]
I'm guessing this is due to the compositor, `picom`, that would be running in a Regolith session but presumably not in a vanilla i3 setup.  killing `picom` would be a quick way of testing this theory.

**Reply by ristomatti** [reply]
@kgilmer Good point, that could be it. I'd try this unless I hadn't already deleted the VM...
